src/shogun/base/SGObject.cpp

```diff
@@ -28,10 +27,46 @@
 
 #include <stdlib.h>
 #include <stdio.h>
-#include <string.h>
+#include <unordered_map>
 
 namespace shogun
 {
+	struct CSGObject::Self
+	{
+
+		Any set(const BaseTag& tag, const Any& any)
+		{
+			Any return_val;
+			if(has(tag))
+				return_val = map[tag];
```

Comment:
Please no space around `=`. Except for logical operators, space is not used between operators and operands in Shogun.